cupy/_core/core.pyx

```diff
@@ -2197,8 +2204,22 @@ cpdef function.Module compile_with_cache(
 
     if prepend_cupy_headers:
         source = _cupy_header + source
+    if jitify:
+        source = '#include <cupy/cuda_workaround.h>\n' + source
     extra_source = _get_header_source()
-    options += ('-I%s' % _get_header_dir_path(),)
+
+    for op in options:
+        if '-std' in op:
```

Comment:
This looks a bit fragile 😅 
User specifying `-I/foo/bar-std/baz` sounds likely.

Reply:
Yeah, I think checking `-std=` (with an extra character `=`) should be more robust?

Reply:
or `-std=c++`...

Reply:
> `-std=c++`

This sounds good to me!
(Strictly speaking `('-std', 'c++11')` is also a valid option but I guess no one do this)

Reply:
Done with [e381c2a](https://github.com/cupy/cupy/pull/7851/commits/e381c2a780573e731090d9d76c19ad0dda5dec52).